Add option to rewrite sway/window title

vrld opened the pull request:
Rewrites window title according to config option "rewrite".

"rewrite" is an object where keys are regular expressions and values are rewrite rules if the expression matches. Rules may contain references to captures of the expression. Regex and replacement follow ECMA-script rules. If no regex matches, the title is left unchanged.

Useful to add icons depending on the window title, e.g.:
```json
"sway/window": {
  "rewrite": {
    "(.*) - Mozilla Firefox": " $1",
    "(.*) - zsh": " [$1]",
  }
}
```

Examples:
- Firefox:
  -  before ![before patch](https://i.imgur.com/LmBSMUy.png)
  -  patched ![after patch](https://i.imgur.com/mhSCDGr.png)
- Terminal / zsh: 
  - before ![before patch](https://i.imgur.com/0RSj65m.png)
  - patched ![after patch](https://i.imgur.com/Ev0v2U9.png)

vrld commented:
Failed test seems to be caused by the runner, not by the changed code:

> ERROR: Clock skew detected. File /Users/runner/work/Waybar/Waybar/build/../subprojects/date-3.0.0/meson_options.txt has a time stamp 10.6111s in the future.

vrld commented:
@Alexays is anything missing?

Alexays commented:
LGTM, can you update the man in order to add the change?
Thanks!

Synthetica9 commented:
Very cool, using this locally and it's working like a charm! +1 for this feature.

Synthetica9 reviewed (changes requested):
Actually, I'd like to request 1 change: if the regex is malformed in some way, it's probably better to keep the window title unchanged rather than crapping out completely and not setting the window title at all as is currently the case.

vrld commented:
@Synthetica9 Good catch. Added try/catch around the regex stuff to fix this issue

@Alexays Added documentation in manpage and rebased on current master.

Alexays commented:
Thx! ❤️ 
Can you also update the Github wiki?

vrld commented:
@Alexays done :)
